src/broker.js

```diff
@@ -110,59 +110,65 @@ module.exports = async (config) => {
 
     // Set up reporting to Streamr stream
     let client
-    let streamId
-    let apiKey
-    if (config.reporting.streamr) {
-        streamId = config.reporting.streamr.streamId
-        apiKey = config.reporting.streamr.apiKey
-        logger.info(`Starting StreamrClient reporting with apiKey: ${apiKey} and streamId: ${streamId}`)
 
-        client = new StreamrClient({
-            auth: {
-                apiKey
-            },
-            autoConnect: true
-        })
-    } else {
-        logger.info('StreamrClient reporting disabled')
+    const streamIds = {
+        metricsStreamId: null,
+        secStreamId: null,
+        minStreamId: null,
+        hourStreamId: null,
+        dayStreamId: null
     }
-
-    let volumeLogger
-    if (config.reporting.perNodeMetrics && config.reporting.perNodeMetrics.enabled) {
-        // set up stream-specific reporting metrics
+    if (config.reporting.streamr || (config.reporting.perNodeMetrics && config.reporting.perNodeMetrics.enabled)) {
         client = new StreamrClient({
             auth: {
-                privateKey: config.ethereumPrivateKey
+                privateKey: config.ethereumPrivateKey,
             },
-            url: config.reporting.perNodeMetrics.wsUrl,
-            restUrl: config.reporting.perNodeMetrics.httpUrl
-            // autoConnect: false
+            url: config.reporting.perNodeMetrics.wsUrl || null,
```

Comment:
Isn't it so that `config.reporting.perNodeMetric` can be set to null? If so, then need to check here that it is not null, i.e., `config.reporting.perNodeMetrics ? (config.reporting.perNodeMetrics.wsUrl || null) : null`